code/game/objects/items/weapons/RCD.dm

```diff
@@ -161,7 +167,77 @@ RCD
 							qdel(A)
 							return 1
 					return	0
+				
+				if(istype(A, /obj/structure/window)) // You mean the grille of course, do you?
+					A = locate(/obj/structure/grille) in A.loc
+				
+				if(istype(A, /obj/structure/grille))
+					if(!checkResource(2, user))
+						return 0
+					to_chat(user, "Deconstructing window...")
+					playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+					if(!do_after(user, 20, target = A))
+						return 0
+					if(locate(/obj/structure/window/full/shuttle) in A.contents)
+						return 0 // Let's not give shuttle-griefers an easy time.
+					if(!useResource(2, user))
+						return 0
+					activate()
+					var/turf/T1 = get_turf(A)
+					qdel(A)
+					A = null
+					for(var/obj/structure/window/W in T1.contents)
+						W.disassembled = 1
+						W.density = 0
+						qdel(W)
+					for(var/cdir in cardinal)
+						var/turf/T2 = get_step(T1, cdir)
+						if(locate(/obj/structure/window/full/shuttle) in T2.contents)
+							continue // Shuttle windows? Nah. We don't need extra windows there.
+						if(!(locate(/obj/structure/grille) in T2.contents))
+							continue
+						for(var/obj/structure/window/W in T2.contents)
+							if(W.dir == turn(cdir, 180))
+								W.disassembled = 1
+								W.density = 0
+								qdel(W)
+						var/obj/structure/window/reinforced/W = new(T2)
+						W.dir = turn(cdir, 180)
+					return 1
 				return 0
+			if(4)
+				if(istype(A, /turf/simulated/floor))
+					if(locate(/obj/structure/grille) in contents)
+						return 0 // We already have window
+					if(!checkResource(2, user))
+						return 0
+					to_chat(user, "Constructing window...")
+					playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+					if(!do_after(user, 20, target = A))
+						return 0
+					if(locate(/obj/structure/grille) in A.contents)
+						return 0 // We already have window
+					if(!useResource(2, user))
+						return 0
+					activate()
+					new /obj/structure/grille(A)
+					for(var/obj/structure/window/W in contents)
+						W.disassembled = 1 // Prevent that annoying glass breaking sound
+						W.density = 0
+						qdel(W)
+					for(var/cdir in cardinal)
+						var/turf/T = get_step(A, cdir)
+						if(locate(/obj/structure/grille) in T.contents)
+							for(var/obj/structure/window/W in T.contents)
+								if(W.dir == turn(cdir, 180))
+									W.disassembled = 1
+									W.density = 0
+									qdel(W)
+						else // Build a window!
+							var/obj/structure/window/reinforced/W = new(A)
+							W.dir = cdir
+					A:ChangeTurf(/turf/simulated/floor/plating) // Platings go under windows.

```

Comment:
No colons unless it's something that's going to give a large performance boost (for example, anything looping through giant lists). Typecast this properly and use dot operators.